total_scattering/autogrouping/autogrouping.py

```diff
@@ -0,0 +1,466 @@
+from sklearn.decomposition import PCA
+from sklearn.metrics import silhouette_score
+from sklearn.preprocessing import StandardScaler
+
+from total_scattering.autogrouping.utils import \
+    peakfitting, \
+    gather_fitparameters, \
+    gathered_parameters_to_tablewksp, \
+    display_parameter_stats, \
+    similarity_matrix_degelder, \
+    similarity_matrix_crosscorr, \
+    get_detector_mask
+from total_scattering.reduction.total_scattering_reduction import compress_ints
+
+
+def plot_features(data, peaks, colnames, labels=None, centroids=None):
+    """
+    Creates a grid plot of the fitted peak parameters against all other params
+    for each peak
+    :param data: array of parameters for each peak, see gather_fitparameters()
+    :param peaks: array of diamond peak positions
+    :param colnames: list of parameter names
+    :param labels: optional labels array from clustering used to color data
+    :param centroids: optional centroids from clustering algorithm
+    :return:
+    """
+    n = len(colnames)
+    npeaks = len(peaks)
+    for p in range(npeaks):
+        fig, ax = plt.subplots(n, n)
+
+        fig.suptitle("peak {}".format(peaks[p]))
+
+        # Plot each peak feature against its other features
+        for i in range(n):
+            for j in range(n):
+                if labels is not None:
+                    ax[i, j].scatter(data[..., p * n + i],
+                                     data[..., p * n + j],
+                                     c=labels, alpha=0.5)
+                else:
+                    ax[i, j].scatter(data[..., p * n + i],
+                                     data[..., p * n + j])
+
+                if centroids is not None:
+                    ax[i, j].scatter(centroids[..., p * n + i],
+                                     centroids[..., p * n + j],
+                                     marker="X", c=np.unique(labels),
+                                     s=200, alpha=0.75,
+                                     edgecolors="red")
+                ax[i, j].set_xlabel(colnames[i])
+                ax[i, j].set_ylabel(colnames[j])
+
+    return fig, ax
+
+
+def get_cachename(prefix, cache_dir, props):
+    filename, sig = CreateCacheFilename(
+        OtherProperties=props, Prefix=prefix, CacheDir=cache_dir)
+    return filename, sig
+
+
+def get_key(key, config):
+    '''Returns the value of key in config. Raises error if not found.'''
+    value = config.get(key, None)
+    if value is None:
+        raise RuntimeError("Expected '{}' to be set in inputs".format(key))
+    return value
+
+
+def get_grouping_method(grouping):
+    '''Make sure grouping is one of the supported options'''
+    method = grouping.split("_")
+    clusterings = ["DBSCAN", "KMEANS"]
+    methods = ["DG", "CC", "ED"]
+
+    if len(method) == 2 and method[0] in clusterings and method[1] in methods:
+        return method
+    else:
+        raise ValueError("Invalid grouping method")
+
+
+def Autogrouping(config):
```

Comment:
This function name breaks the [Python naming conventions](https://namingconvention.org/python/). If we switch to `autogrouping`, that will comply.